### PR TITLE
Fix reference links in poisson_conf_interval API docs

### PR DESCRIPTION
**Pull request by larrybradley**
This PR fixes the external reference links in the `poisson_conf_interval` API docs.  Currently the external links are not rendered correctly.

### CODE REVIEW

File: astropy/stats/funcs.py [hunk 1]
```diff
@@ -577,20 +586,22 @@ def poisson_conf_interval(n, interval='root-n', sigma=1, background=0,
...
-    [KraftBurrowsNousek][kbn1991] discuss this choice in more detail and show
-    that the problem is relatively insensitive to the choice of prior.
+    `Kraft, Burrows, and Nousek (1991)
+    <http://adsabs.harvard.edu/abs/1991ApJ...374..344K>`_ discuss this
+    choice in more detail and show that the problem is relatively
+    insensitive to the choice of prior.
 
-    This functions has an optional dependency: Either scipy or
-    `mpmath <http://mpmath.org/>`_  need to be available. (Scipy only works for
+    This functions has an optional dependency: Either scipy or `mpmath
```
Comment (pllim) [inline astropy/stats/funcs.py]: Can Sphinx understand line break in the middle of the linking like this?

Comment (larrybradley) [reply astropy/stats/funcs.py]: It seems to.  The docs rendered fine locally.

I just pushed a change to fix typos I noticed in this line. :)

**Comment by larrybradley** [comment]
These links all work.  However, running `linkcheck` reveals several broken links.

**Comment by pllim** [comment]
Hmm! I thought we have `linkcheck` running in `cron` but now that I look at the log, it seems weird... https://travis-ci.org/astropy/astropy/jobs/575825712 @bsipocz , you have idea what is happening?

File: astropy/stats/funcs.py [hunk 2]
```diff
@@ -577,20 +586,22 @@ def poisson_conf_interval(n, interval='root-n', sigma=1, background=0,
...
-
-    This functions has an optional dependency: Either scipy or
-    `mpmath <http://mpmath.org/>`_  need to be available. (Scipy only works for
-    N < 100).
+    `Kraft, Burrows, and Nousek (1991)
+    <http://adsabs.harvard.edu/abs/1991ApJ...374..344K>`_ discuss this
+    choice in more detail and show that the problem is relatively
+    insensitive to the choice of prior.
 
+    This function has an optional dependency: Either scipy or `mpmath
```
[the remark this answers is not in the document]
Comment (larrybradley) [reply astropy/stats/funcs.py]: I didn't write that sentence, but I just fixed it.

**Pull request merged**